08.ls_object/lib/command.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+require_relative 'path'
+
+module LS
+  class Command
+    COLUMN_COUNT = 3
+
+    def initialize(dotmatch: false, reverse: false, long_format: false)
+      @dotmatch = dotmatch
+      @reverse = reverse
+      @long_format = long_format
+    end
+
+    def exec
+      collected_paths = collect_paths
+      sorted_paths = sort_paths(collected_paths)
+      paths = sorted_paths.map { |path| Path.new(path) }
+      @long_format ? list_long(paths) : list_short(paths)
+    end
+
+    private
+
+    def sort_paths(collected_paths)
+      @reverse ? collected_paths.reverse : collected_paths
+    end
+
+    def collect_paths
+      @dotmatch ? Dir.glob('*', File::FNM_DOTMATCH) : Dir.glob('*')
+    end
+
+    def list_long(paths)
+      blocks = paths.sum(&:blocks)
+      total = "total #{blocks}"
+      body = render_long_format_body(paths)
+      puts [total, *body]
+    end
+
+    def render_long_format_body(paths)
+      max_sizes = find_max_sizes(paths)
+      paths.map { |path| format_row(path, max_sizes) }
+    end
+
+    def find_max_sizes(paths)
+      {
+        nlink: paths.map { |path| path.nlink.to_s.length }.max,
```

Comment:
```suggestion
        nlink: paths.map(&:nlink).max.to_s.length,
```

とすると、ループ内でのメソッド呼び出しが減るので効率がよくなりそうです。（49行目も同様）